test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -5962,10 +5854,14 @@ def submit(self, obj, script=None, extend=None, submit_dir=None):
             return None
 
         if submit_dir is None:
```

Comment:
you can replace this if block as follow
```py
if not submit_dir and self.du.get_platform() != 'shasta':
   submit_dir = pwd.getpwnam(obj.username)[5]
```

Reply:
I realized that, this will fail if platform is non-shasta and submit_dir is not none, then it will go in else condition and assigns submit_dir to None which is incorrect. 
Proper if-else condition would be 
```suggestion
       if submit_dir and self.du.get_platform == 'shasta':
            submit_dir = None
       else:
            submit_dir = pwd.getpwnam(obj.username)[5]
```
Agree ?

Reply:
@shilpakodli I was talking about replacing if including internal if-else block. So we want to modify submit dir only when submit_dir is None and platform != 'shasta' right? which `if not submit_dir and self.du.get_platform() != 'shasta':` condition will do